src/parser/GenericParser.test.ts

```diff
@@ -0,0 +1,281 @@
+        jest.spyOn(MetricCalculator.prototype, "calculateMetrics").mockImplementation(
+            async (parsedFilePromise) => {
+                await parsedFilePromise;
+                throw new Error("Buuuh!");
+            }
+        );
+
+        const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {
+            /* Do nothing */
+        });
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid/path.cpp"));
+        const actualResult = await parser.calculateMetrics();
+
+        // TODO: do no longer include errors into the list of results, use "info" field instead.
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([["clearly/invalid/path.cpp", expectedFileMetricsMap]])
+        );
+        expect(errorSpy).toHaveBeenCalled();
+    });
+
+    it("should perform the appropriate actions in case there is an error for only the first of two files while calculating metrics", async () => {
+        const expectedFileMetricsMap = new Map([
+            [FileMetric.linesOfCode, { metricName: FileMetric.linesOfCode, metricValue: 5 }],
+        ]);
+        const expectedErrorMetricsMap = new Map([
+            ["ERROR", { metricName: "ERROR", metricValue: -1 }],
+        ]);
+
+        const findFilesAsyncMock = findFilesAsync as jest.Mocked<typeof findFilesAsync>;
+        findFilesAsyncMock.mockImplementation((config) => {
+            return mockedFindTwoFiles();
+        });
+
+        const treeParserSpied = jest
+            .spyOn(TreeParser, "parse")
+            .mockImplementation((file, assumedLanguage) => {
+                return Promise.resolve({
+                    fileExtension: file.fileExtension,
+                    filePath: file.filePath,
+                    language: Language.CPlusPlus,
+                    tree: tree,
+                });
+            });
+
+        const calculateMetricsSpied = jest
+            .spyOn(MetricCalculator.prototype, "calculateMetrics")
+            .mockImplementation(async (parsedFilePromise) => {
+                const file = await parsedFilePromise;
+                if (file.fileExtension !== "cpp") {
+                    throw new Error("I only accept cpp files!");
+                }
+                return [file.filePath, expectedFileMetricsMap];
+            });
+
+        const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {
+            /* Do nothing */
+        });
+
+        const parser = new GenericParser(getParserConfiguration("clearly/invalid"));
+        const actualResult = await parser.calculateMetrics();
+
+        expect(actualResult.fileMetrics).toEqual(
+            new Map([
+                ["clearly/invalid/path1.cc", expectedErrorMetricsMap],
+                ["clearly/invalid/path2.cpp", expectedFileMetricsMap],
+            ])
+        );
+        expect(errorSpy).toHaveBeenCalled();
+    });
+
+    it("should fail gracefully if findFilesAsync throws an error", async () => {
```

Comment:
A general note: If there is a lot of copied test code: Tests should be handled like production code. Let's create helper methods if it is helpful, let's move reusable code into beforeEach methods, etc.